scenarios/remote_signals/src/main.rs

```diff
@@ -138,15 +136,22 @@ fn agent_behaviour(
                     let admin_client = AdminWebsocket::connect(admin_ws_url, reporter).await?;
                     // No more agents available to signal, get a new list.
                     // This is also the initial condition.
-                    let mut new_peer_list = admin_client
+                    let agent_infos_encoded = admin_client
                         .agent_info(None)
                         .await
-                        .context("Failed to get agent info")?
+                        .context("Failed to get agent info")?;
+                    let mut agent_infos = Vec::new();
+                    for info in agent_infos_encoded {
+                        let a = kitsune2_api::AgentInfoSigned::decode(
+                            &kitsune2_core::Ed25519Verifier,
+                            info.as_bytes(),
+                        )?;
+                        agent_infos.push(AgentPubKey::from_raw_32(a.agent.to_vec()))
+                    }
+                    let new_peer_list = agent_infos
                         .into_iter()
-                        .map(|info| AgentPubKey::from_raw_36(info.agent.0.clone()))
-                        .filter(|k| k != cell_id.agent_pubkey()) // Don't signal ourselves!
+                        .filter(|k| k != cell_id.agent_pubkey()) // Don't call ourselves!
                         .collect::<Vec<_>>();
-                    new_peer_list.shuffle(&mut thread_rng());
```

Comment:
Why delete this? It's part of the logic of the test?

Reply:
whoops, copy/paste mistake.

Reply:
[fixed](85dca7894b9a48407b87a78e656f5070346b5aea)